packages/yoroi-ergo-connector/src/inject.js

```diff
@@ -192,17 +210,33 @@ if (shouldInject()) {
     });
 
     // events from page (injected code)
-    window.addEventListener("message", function(event) {
+    window.addEventListener("message", async function(event) {
         if (event.data.type === "connector_rpc_request") {
             console.log("connector received from page: " + JSON.stringify(event.data) + " with source = " + event.source + " and origin = " + event.origin);
             yoroiPort.postMessage(event.data);
         } else if (event.data.type == "connector_connect_request") {
             // URL must be provided here as the url field of Tab is only available
             // with the "tabs" permission which Yoroi doesn't have
-            yoroiPort.postMessage({
-                type: "yoroi_connect_request",
-                url: location.hostname
-            });
+            try {
+                if (location.hostname === 'localhost') {
+                    yoroiPort.postMessage({
+                        imgBase64Url: '',
+                        type: "yoroi_connect_request",
+                        url: location.hostname
+                    });
+                    return;
+                }
+                const faviconURL = `https://services.keeweb.info/favicon/${location.hostname}`
```

Comment:
The connector already has access to the HTML for the page, so I don't think we need this kind of service. We should be able to do the following:

1. Check if an icon exists in the HTML (`document.querySelector("link[rel~='icon']")`)
2. If not, check if a favicon exists at the document root

Reply:
I tried to do this way but I got CORS error when getting the image to parse it, even when I set `https://*` in permissions `manifest.json`

Reply:
The code to query the favicon has to be done in the context of the website and not in the context of the extension. The extension can't make requests to other websites, but the page we're injecting code into should be able to.

If the website you're injecting code into has its own CORS policy that is blocking it, that is a separate issue (and entirely possible, but I would think most websites allow querying their own favicon)

Reply:
@SebastienGllmt I found that cross site request are blocked in content script (context of website) as of chrome 73, even when we allow them in permissions property.  So we need to make the request in the background script. I've just made it, and make it async. Let me know if that works for you 🙌